compiler/src/dotty/tools/dotc/transform/OverridingPairs.scala

```diff
@@ -215,15 +216,16 @@ object OverridingPairs:
             }
       )
     else
-      // releaxed override check for explicit nulls if one of the symbols is Java defined,
-      // force `Null` being a subtype of reference types during override checking
-      val relaxedCtxForNulls =
+      def matchNullaryLoosely = member.matchNullaryLoosely || other.matchNullaryLoosely || fallBack
```

Comment:
I think this would be clearer if it followed the pattern in Types.scala matches:

```scala
val overrideCtx = if ... then ... else ctx
def matchNullaryLosely = ...
memberTp.overrides(...)(using overrideCtx)
```

Reply:
Done